services/hidadapter.md

```diff
@@ -0,0 +1,87 @@
+# HID Adapter Configurator
+
+    identifier: 0x1e5758b5
+    
+A service for configuring how Jacdac device map to HID input events. Users can have multiple configurations and swap between them using the `swapTo` or `next` commands.
+
+## Registers
+
+    rw num_configurations: u8 @ 0x80
+    
+The number of configurations stored on the server.
+
+    rw current_configuration: u8 @ 0x81
+
+The current configuration the server is using.
+
+## Commands
+
+    command get_configuration @ 0x80 {
+        results: pipe
+        configuration_number: u8
+    }
+    pipe report configuration {
+        configuration_number: u8
+        binding_index: u8
+        device_id: u8[8]
+        service_class: u32
+        service_idx: u8
+        trigger_context:u8
+        trigger_value: u32
+        selector: u16
+        modifiers:u16
+    }
+    
+Retrieves a configuration stored on the server. If the configuration does not exist, an empty report will be returned
+
+    command set_configuration @ 0x80 {
```

Comment:
this requires that there is a report from set_configuration that gives pipe_port, and then the client writes to that port, see https://github.com/microsoft/jacdac/blob/main/services/modelrunner.md#commands